data_quality_pipeline/src/made/data_pipeline/data/datacomp_handler.py

```diff
@@ -64,6 +67,56 @@ def filter_by_uid(sample):
         .batched(batch_size)
     )
 
+
+def decode_webdataset_two_steps(
```

Comment:
This implementation looks good to me, I think you can rename the function to `decode_webdataset` and delete the old implementation.

Please, run and update (if needed) the tests `data_quality_pipeline/test/test_webdataset_reader.py`